src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Allows for the comparison of two <see cref="ISymbol"/> instance based on certain criteria TODO
+    /// </summary>
+    public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol>
+    {
+        /// <summary>
+        /// Compares two <see cref="ISymbol"/> instances based on the default comparison rules, equivalent to calling <see cref="IEquatable{ISymbol}.Equals(ISymbol)"/>
+        /// </summary>
+        public static readonly SymbolEqualityComparer Default = new SymbolEqualityComparer(TypeCompareKind.AllNullableIgnoreOptions);
+
+        /// <summary>
+        /// Compares  two <see cref="ISymbol"/> instances, considering their nullability
+        /// </summary>
+        public static readonly SymbolEqualityComparer IncludeNullability = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything2); //TODO: should this be explicitly *not* compare everything
```

Comment:
>ConsiderEverything2 [](start = 118, length = 19)

Shouldn't use 2

Reply:
We've been using ConsiderEverything2 to track instances of where we need to go back and think about the usage. This is one of those. We're saying (for now) that including nullability == ConsiderEverything, but it probably should be explicit so that adding comparisons in the future doesn't change its meaning.

---
In reply to: [305121678](https://github.com/dotnet/roslyn/pull/37247#discussion_r305121678) [](ancestors = 305121678)